pytest/tests/mocknet/utils.py

```diff
@@ -17,4 +18,98 @@ def get_systemd_time_spec(self):
 
 class ScheduleContext(BaseModel):
     id: str
-    schedule: ScheduleMode
\ No newline at end of file
+    schedule: ScheduleMode
+
+
+### Stake distribution ###
+class StakeDistribution:
+
+    def __init__(self, num_chunk_producers):
+        self.num_chunk_producers = num_chunk_producers
+        self.producer_idx = 0
+
+    def get_next_validator_stake(self):
+        pass
+
+    def get_next_producer_stake(self):
+        pass
+
+    def get_stake(self, node):
+        if node.role(
+        ) == 'validator' or self.producer_idx >= self.num_chunk_producers:
+            amount = self.get_next_validator_stake()
+        else:
+            amount = self.get_next_producer_stake()
+        return amount
+
+
+class MainnetStakeDistribution(StakeDistribution):
+    """
+    Uses the mainnet validator stakes as the stake distribution.
+    To avoid going below the seat price, the stake of the bottom `double_after` validators is doubled.
+    If more validators are requested than available, the stake of the last validator is used for the remaining validators.
+    To avoid mainnet stake leaking to the mocknet, the stake is amplified by `multiplier`.
+    """
+
+    @staticmethod
+    def get_mainnet_stakes():
+        """
+        Makes an RPC call to mainnet to get current validator stakes
+        """
+        url = "https://rpc.mainnet.near.org"
+        payload = {
+            "jsonrpc": "2.0",
+            "id": "dontcare",
+            "method": "validators",
+            "params": [None]
+        }
+        response = requests.post(url, json=payload)
+        if response.status_code == 200:
+            data = response.json()
+            return [
+                int(v['stake']) for v in data['result']['current_validators']
+            ]
+        else:
+            raise Exception(
+                f"Failed to get mainnet stakes: {response.status_code}")
+
+    def __init__(self, num_chunk_producers):
+        super().__init__(num_chunk_producers)
+        self.mainnet_stakes = MainnetStakeDistribution.get_mainnet_stakes()
+        self.num_validators = len(self.mainnet_stakes)
+        self.validator_idx = self.num_chunk_producers
+        self.double_after = 100
```

Comment:
I think it's better to multiply bottom-100 stakes by 2 right here, instead of handling it in `get_next_validator_stake`. Then we don't need `double_after`.
We can also sort stakes after that operation again, to keep them sorted.
Same logic goes to `multiplier`. If we don't plan to try other multipliers, can simply define a static constant here (is it possible?) and use it in constructor.

More global comment:
I think there is some convention to start fields from underscore (`_validator_idx`) to state that they are designed for internal use only.
By not doing that, the author states that field can be accessed and modified arbitrarily outside of the class.
Perhaps we should prepend many fields across the codebase with `_`.